state/protocol/badger/mutator.go

```diff
@@ -443,10 +473,18 @@ func (m *FollowerState) insert(ctx context.Context, candidate *flow.Block, last
 			return fmt.Errorf("could not store candidate block: %w", err)
 		}
 
-		qc := candidate.Header.QuorumCertificate()
-		err = m.qcs.StoreTx(qc)(tx)
-		if err != nil && !errors.Is(err, storage.ErrAlreadyExists) {
-			return fmt.Errorf("could not store qc: %w", err)
+		if !qcAlreadyInserted {
+			err = m.qcs.StoreTx(qc)(tx)
+			if err != nil {
+				return fmt.Errorf("could not store incorporated qc: %w", err)
+			}
+		}
+
+		if certifyingQC != nil {
+			err = m.qcs.StoreTx(certifyingQC)(tx)
+			if err != nil {
+				return fmt.Errorf("could not store certifying qc: %w", err)
+			}
 		}
```

Comment:
```suggestion
		if certifyingQC != nil {
			err = m.qcs.StoreTx(certifyingQC)(tx)
			if err != nil {
				if !errors.Is(err, storage.ErrAlreadyExists) {
					return fmt.Errorf("could not store incorporated qc: %w", err)
				}
			} else {
				events = append(events, func () { m.consumer.BlockProcessable(candidate.Header, certifyingQC) })
			}
		}
```

Reply:
I don't think there is a case where we will have the certifying QC already inserted, it means that we have two parents for one child which is impossible. 

QCs are inserted in `Extend` and `ExtendCertified`. `Extend` adds QC certifying parent, `ExtendCertified` adds QC for itself and the most important thing is that we adds blocks only if parent is known meaning we couldn't get an `ErrAlreadyExists` for certifying QC.

Reply:
https://github.com/onflow/flow-go/pull/3949/commits/a42afb0d3ad5d09d473b81730b0c13e41fd310b8

Reply:
>  it means that we have two parents for one child which is impossible.

👍 Agree - good point